### DISCUSSION TITLE
seeder関係の質問

### DISCUSSION THREAD

**Discussion by takahashi123456** [ccc-2021/app#36] [Q&A] [answered]
@kai0310 
お疲れ様です。今、コードを見直してたところ、seedしてるとエラーが出たので報告します。

app/app/Models/Schedule.phpの

```PHP
public function getRateAvgAttribute(): int
     {
         return $this->reviews?->avg('rate') ?? 0;
     }
```

の?の部分ってtypoですか？
あと、
ReviewSeeder.phpの

```PHP
$schedules->each(function ($schedule) use ($users) {
            Review::factory()
                ->count(5)
                ->state(new Sequence(
                    fn() => [
                        'user_id' => $users->random()->id,
                        'schedule_id' => $schedule->id,
                    ],
                ))
                ->create();
        });
```

の部分 fn() => でエラーが出て動かなくなってます。
直そうと思ったんですが、アロー関数がよくわからなくて直せませんでした。
直していただけると幸いです。

**Comment by kai0310** [comment, marked as answer]
> の?の部分ってtypoですか？

null safe operator という構文です。

---
文法的なエラーは見つけられなかったため、エラーメッセージを提示してください。

**Reply by kai0310** [reply]
https://github.com/ccc-2021/app/blob/f3c83977d95e9c99a7ce5700249cf2522cf5f180/database/seeders/DatabaseSeeder.php#L26

関係ないですが、これいらないので消しちゃってください🙏

**Comment by takahashi123456** [comment]
@kai0310 
ありがとうございます。こちらの環境のなんかのエラーなんですかね。。
こんな感じのエラーが出てます

```
   ParseError 

  syntax error, unexpected '=>' (T_DOUBLE_ARROW), expecting ')'

  at database/seeders/ReviewSeeder.php:28
     24▕         $schedules->each(function ($schedule) use ($users) {
     25▕             Review::factory()
     26▕                 ->count(5)
     27▕                 ->state(new Sequence(
  ➜  28▕                     fn() => [
     29▕                         'user_id' => $users->random()->id,
     30▕                         'schedule_id' => $schedule->id,
     31▕                     ],
     32▕                 ))

      +1 vendor frames 
  2   [internal]:0
      Composer\Autoload\ClassLoader::loadClass("Database\Seeders\ReviewSeeder")

  3   [internal]:0
      spl_autoload_call("Database\Seeders\ReviewSeeder")
```

**Reply by kai0310** [reply]
帰宅したら確認しておきます！
ここの問題は無視しておいてください👏

**Reply by kai0310** [reply]
ちなみに実行コマンドはなんですか？
`./vendor/bin/sail artisan migrate:fresh —seed`
ですか？
その場合、PHPのバージョンを確認して欲しいです。
`./vendor/bin/sail php -v`

**Reply by takahashi123456** [reply]
コマンドは　php artisan migrate:fresh --seed　です。

PHP 7.3.24-(to be removed in future macOS) (cli) (built: Jun 17 2021 21:41:13) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.24, Copyright (c) 1998-2018 Zend Technologies

になってます。。
お手数おかけします。。

**Reply by kai0310** [reply]
PHP 8系を使っていないことが原因かと思います。

**Reply by kai0310** [reply]
Dockerでは、8系が用意されているかと思うのでそちらを使ってください…

**Reply by takahashi123456** [reply]
りょかいです。。

**Reply by kai0310** [reply]
```
./vendor/bin/sail artisan migrate:fresh --seed
```

**Reply by takahashi123456** [reply]
データベースとの接続でエラーが出てしまうのでまた後で教えていただいても良いですか・・

**Reply by kai0310** [reply]
こちらを参考にしてください

https://zenn.dev/kai0310/articles/53ff068f8e19b7

**Reply by takahashi123456** [reply]
いけました！！！
助かりました！！！

**Reply by kai0310** [reply]
Seedingがですか？

**Reply by takahashi123456** [reply]
かいさんの記事のやつ試してみてseedingできました。☺️
